interp/src/values.rs

```diff
@@ -161,12 +161,64 @@ impl Value {
     pub fn as_u64(&self) -> u64 {
         let mut val: u64 = 0;
         for (index, bit) in self.vec.iter().by_ref().enumerate() {
-            val += u64::pow(2, (index as usize).try_into().unwrap())
-                * (*bit as u64);
+            if *bit {
+                //protects against panic in case of # less than u64::max in
+                // value of width greater than 64
+                val += u64::pow(2, (index as usize).try_into().unwrap())
+                    * (*bit as u64);
+            }
+        }
+        val
+    }
+
+    /// Converts value into u128 type. Vector within Value can be of any width.
+    ///
+    /// # Example
+    /// ```
+    /// use interp::values::*;
+    /// let unsign_128_16 = (Value::try_from_init(16, 16).unwrap()).as_u128();
+    /// ```
+    pub fn as_u128(&self) -> u128 {
+        let mut val: u128 = 0;
+        for (index, bit) in self.vec.iter().by_ref().enumerate() {
+            val += u128::pow(2, (index as usize).try_into().unwrap())
+                * (*bit as u128);
         }
         val
     }
 
+    /// Converts value into i64 type using 2C representation.
+    /// # Example
+    /// ```
+    /// use interp::values::*;
+    /// let signed_neg_1_4 = (Value::try_from_init(15, 4).unwrap()).as_i64();
+    /// assert_eq!(signed_neg_1_4, -1);
+    /// ```
+    pub fn as_i64(&self) -> i64 {
+        let vec_len = self.vec.len() as u32;
```

Comment:
Why is this value being cast to a `u32`? Seems better to keep it as a `usize`

Reply:
I did that because `i64::pow` needs a u32 as `exp`

Reply:
Ah gotcha